src/SFA.DAS.PR.Application.UnitTests/Requests/Commands/DeclinedRequest/DeclinedRequestCommandHandlerTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Application.Requests.Commands.CreatePermissionRequest;
+using SFA.DAS.PR.Application.Requests.Commands.DeclinedRequest;
+using SFA.DAS.PR.Data.Repositories;
+using SFA.DAS.PR.Data.UnitTests.InMemoryDatabases;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.PR.Application.UnitTests.Requests.Commands.DeclinedRequest;
+
+public sealed class DeclinedRequestCommandHandlerTests
+{
+    [Test]
+    [RecursiveMoqAutoData]
+    public async Task Handle_DeclinedRequestCommand_Successful(
+        Request request,
+        DeclinedRequestCommand command
+    )
+    {
+        request.Id = command.RequestId;
+
+        ValidatedResponse<Unit> result = null!;
+
+        Request? updatedRequest = null;
+
+        using (var context = InMemoryProviderRelationshipsDataContext.CreateInMemoryContext(
```

Comment:
A unit test should only know about the unit it is testing, here the repository internals have been leaked which is a dependency, so if we change the repository implementation to use document database, we will have to rewrite this ~unit~ test. Here we should have just verified the repository call that way if repository implementation changes, this test is not impacted.